app/controllers/posts_controller.rb

```diff
@@ -2,23 +2,41 @@
 
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    top_post_id = Post.joins(:votes).group(:post_id).count.sort.to_h
```

Comment:
tutaj kierunek był dobry, ale trochę można to poprawić

ok, mamy posty połączone z głosami i pogrupowane po postach
```
Post.joins(:votes).group(:post_id)
```

Metoda count zwraca nam hash z danymi "id_posta" => "liczba komentarzy" i wszystkie kolejne operacje już dzieją się poza bazą danych czyli powoli. Nie ma to znaczenia przy 10 czy 100 postach, ale przy 100 000, albo 1 000 000 będzie miało ogromne.
```
Post.joins(:votes).group(:post_id).count
```

kolejna rzecz - nie jest uwzględnione czy oceny są pozytywne czy negatywne

żeby zachować obecną funkcjonalność (bez uwzględniania negatywnych ocen) można użyć `order()`, który będzie jeszcze w bazie danych
```
Post.joins(:votes).group(:post_id).order('count(post_id) desc')
```

żeby `order()` działał to musimy kolumnę po której sortujemy wybrać jako wynik zapytania używając `select()`
```
Post.joins(:votes).group(:post_id).select('count(post_id) as count').order('count desc')
```

i jeszcze żeby zwrócić ID też musimy dodać je do select, ale także do group by
```
Post.joins(:votes).group(:post_id, :id).select('posts.id, count(post_id) as count').order('count desc').last.id
```